### ISSUE TITLE
strict_variables not throwing exception for undefined object properties

### ISSUE THREAD

**Issue by scottgeye** [harttle/liquidjs#75]
Parsing with strict_variables seems to do what it should if one of my value variables is undefined (e.g. {{name}}), but it doesn't throw an exception if my variable is an object referencing a property (e.g. {{store.name}})

Here's some example code.  

```
const data = {store:{}};
const Liquid = require('liquidjs');
const liquid = Liquid({ strict_variables: true, strict_filters: true });
liquid.parseAndRender('{{store.name}}', data);
```

All of my variables are nested like this, so it would be super useful to know if these properties are undefined without having to manually check.

Thanks

**Comment by harttle** [comment]
fixed on liquidjs@5.0.1

**Comment by SergeL** [comment]
Hi, unfortunately it throws now also when `strict_variables` is set to `false`.

In our case it is important to disable this case since we're handling not available values manually like:

```
Hello {% if user.given_name %}{{ user.given_name }}{% else %}Friend{% endif %},<br/>
```
